xbmc/cores/VideoPlayer/VideoRenderers/VideoShaders/ConversionMatrix.cpp

```diff
@@ -455,49 +490,53 @@ void CConvertMatrix::GenMat()
   }
 
   ConversionToRGB mConvRGB(convYCbCr.Kr, convYCbCr.Kb);
-
-  m_pMat.reset(new CGlMatrix(mConvRGB.Get()));
+  CGlMatrix mat(mConvRGB);
 
   CTranslate trans(0, -0.5, -0.5);
-  *m_pMat *= trans;
+  mat *= trans;
 
   if (m_limitedSrc)
   {
     if (m_srcBits >= 12)
     {
       CScale scale(4080.0f / (3760 - 256), 4080.0f / (3840 - 256), 4080.0f / (3840 - 256));
       CTranslate trans(- 256.0f / 4080, - 256.0f / 4080, - 256.0f / 4080);
-      *m_pMat *= scale;
-      *m_pMat *= trans;
+      mat *= scale;
+      mat *= trans;
     }
     else if (m_srcBits == 10)
     {
       CScale scale(1020.0f / (940 - 64), 1020.0f / (960 - 64), 1020.0f / (960 - 64));
       CTranslate trans(- 64.0f / 1020, - 64.0f / 1020, - 64.0f / 1020);
-      *m_pMat *= scale;
-      *m_pMat *= trans;
+      mat *= scale;
+      mat *= trans;
     }
     else
     {
       CScale scale(255.0f / (235 - 16), 255.0f / (240 - 16), 255.0f / (240 - 16));
       CTranslate trans(- 16.0f / 255, - 16.0f / 255, - 16.0f / 255);
-      *m_pMat *= scale;
-      *m_pMat *= trans;
+      mat *= scale;
+      mat *= trans;
     }
   }
 
   if (m_srcTextureBits > 8)
   {
     float val = 65535.0f / ((1 << m_srcTextureBits) - 1);
     CScale scale(val, val, val);
-    *m_pMat *= scale;
+    mat *= scale;
   }
+
+  return mat;
 }
 
-void CConvertMatrix::GetYuvMat(float (&mat)[4][4])
+Matrix4 CConvertMatrix::GetYuvMat()
 {
-  if (!m_pMat)
-    return;
+  if (!m_mat.IsInitialized())
+  {
+    m_mat = GenMat();
+    m_mat.SetInitialized();
+  }
```

Comment:
this could be extracted to a method

Reply:
I'm not 100% sure what to do with this yet.

Reply:
Took another look, I think it's fine as-is. I missed the fact that the two occurrences of this pattern are about different matrices.